syncer/datacenter/exclude.go

```diff
@@ -35,19 +35,15 @@ func (s *store) exclude(data *pb.SyncData, allMapping pb.SyncMapping) {
 }
 
 // excludeInstances find out the instance data belonging to self, through the mapping table
-func (s *store) excludeInstances(ins []*scpb.MicroServiceInstance, mapping pb.SyncMapping) []*scpb.MicroServiceInstance {
-	is := make([]*scpb.MicroServiceInstance, 0, len(ins))
-	for _, item := range ins {
-		_, ok := mapping[item.InstanceId]
-		if ok {
-			if item.Status != "UP" {
-				delete(mapping, item.InstanceId)
-			}
+func (s *store) excludeInstances(ins []*scpb.MicroServiceInstance, curMapping pb.SyncMapping) []*scpb.MicroServiceInstance {
+	nis := make([]*scpb.MicroServiceInstance, 0, len(ins))
+	for _, inst := range ins {
+		if index := curMapping.CurrentIndex(inst.InstanceId); index != -1 {
 			continue
 		}
-		if item.Status == "UP" {
-			is = append(is, item)
+		if inst.Status == "UP" {
```

Comment:
I think status is the attribute of the service-center , so it could be moved to the plugin implement.

Reply:
It is not impact the function, we can resolve it in the later PR.